psydac/core/bsplines_kernels.py

```diff
@@ -35,8 +43,14 @@ def find_span_p(knots: 'float[:]', degree: int, x: float):
         Springer-Verlag Berlin Heidelberg GmbH, 1997.
     """
     # Knot index at left/right boundary
+    #In the periodic case the p+1 knots is the last knot equal to the left 
+    #boundary, while the first knots being the right boundary is multiplicity+p
+    #before the last knot of the sequence (see make knots)
```

Comment:
I guess this sentence means 
```
in the periodic case, the last knot located on the left boundary is the one with index i = p
and the first knot located on the right boundary is the one with index i = -m-p
where p = degree and m = multiplicity
```
by the way, the first statement holds also in the non-periodic case.

Reply:
Exactly

Reply:
As for the second statement, I'm not sure to follow the logic: I don't see why the multiplicity matters to the index of the first knot on the right boundary.

Reply:
Consider for example the periodic case with n_cells = 3, degree =2, multiplicity = 2
Then the dimension of the spline space is 6, and the knots needed to evaluate the splines on the domain are:
(-1/3, 0, 0, 1/3, 1/3, 2/3, 2/3, 1, 1, 4/3)
the first knot on the right boundary has index -3 = -degree-1

Reply:
ok, but why adding the second knot 4/3 in the sequence ?
Given that the B-spline defined by the knots (1, 1, 4/3, 4/3) vanishes on the domain [0,1], I don't see the purpose of this

Reply:
I think I remember, we are adding those basis function for compatibiliy with the parallel case where you have p function on the ghost regions. As in the non periodic case you use the ghost regions to enforce the periodicity I think we need those more basis function. 
Is that right @yguclu

Reply:
Yes, and this was designed for the case of `multiplicity == 1`. When the multiplicity is higher we need to repeat fewer extra basis functions on both sides.

In fact one could argue that a smaller ghost region is needed when `multiplicity > 1`. The problem is that the ghost regions are mostly designed with the assembled FEM matrices in mind, where the required ghost regions are larger not smaller.

Reply:
I think that we can forget about the ghost regions for the moment. We should be able to build the smallest knot sequence that is needed for the evaluation of the field, knowing that we store the correct amount of spline coefficients (equal to the dimension of the space).

Reply:
I have implemented the previous solution but it's not working really well, I have really high errors in maxwell 2d (10^14 ...). 
I will continue looking to see if I can understand why it's not working.

Reply:
Since we discussed the precise meaning of this:
```
    # Knot index at left/right boundary
    low  = degree
    high = len(knots)-1-degree
```
I suggest to specify the comment and write
```
    # last knot on the left boundary
    low  = degree
    # first knot on the right boundary
    high = len(knots)-1-degree
```

Reply:
thanks !